prov/efa/src/efa_av.c

```diff
@@ -234,7 +235,22 @@ static int efa_av_insert_ah(struct efa_av *av, struct efa_ep_addr *addr,
 		}
 
 		memcpy(&reverse_av->key, &key, sizeof(key));
-		reverse_av->fi_addr = *fi_addr;
+		if (av->ep_type == FI_EP_RDM) {
+			/*
+			 * RDM endpoint has an extra field in the raw address
+			 * to detect QP reuse, so this lookup cannot return the
+			 * fi_addr_t. Only the upper layer can determine the
+			 * address vector index after validating that extra
+			 * field, so don't set it here.
+			 */
+			/* TODO: don't set fi_addr once the completion queues are combined */
+			//reverse_av->fi_addr = FI_ADDR_NOTAVAIL;
```

Comment:
The comment is good but should not be in this commit.
It basically saying that set `reverse_av->fi_addr` here for RDM endpoint is a bug. But we cannot do it yet because some cq code has to change first.

I moved the comment to a future commit 
`prov/efa: set av_entry->fi_addr to FI_ADDR_NOTAVAIL`
that actually set `reverse_av->fi_addr` to `FI_ADDR_NOTAVAIL`